lib/ansible/plugins/filter/core.py

```diff
@@ -223,6 +223,30 @@ def from_yaml_all(data):
         return yaml_load_all(text_type(to_text(data, errors='surrogate_or_strict')))
     return data
 
+def from_keyvalue(data, delimiter="=", skip_invalid=False):
+    '''Parse key/value pairs into a dict'''
+    regex = re.compile(r'\s*' + delimiter + r'\s*')
+
+    lines = []
+    if (isinstance(data, string_types)):
+        lines = data.splitlines()
+    elif (isinstance(data, list)):
+        lines = data
+    else:
+        raise AnsibleFilterError('Filter only supports string or list types')
+
+    result = {}
+    for line in lines:
+        try:
+            (key, value) = re.split(regex, line, 1)
+            result[key] = value
+        except ValueError as e:
+            if (skip_invalid):
+                pass
+            else:
+                raise AnsibleFilterError('Unable to parse key/value(%s): %s' % (line, to_native(e)))
+
+    return result
```

Comment:
Why reinvent wheel and not use  the existing common functions (parse_kv, split_args?)

Reply:
I did not know about them and doing a quick search on google for them don't see any documentation about them.

Reply:
https://github.com/ansible/ansible/blob/devel/lib/ansible/parsing/splitter.py

also, you probably want to add these in a collection, not core

Reply:
Looking at parse_kv it does not allow changing the delimeter.  I know it might not be a common use case but I also needed to parse `key: value` data that was not json from the results of running a command.  Would that be achievable using parse_kv?

Reply:
from_yaml should do that easily

Reply:
Damn, I didn't even thing about using `from_yaml`.  It does work for my case.

But still is there a filter or simple way to process key/value pairs using a filter?  I can see the benefit of having one if there is not and it should  be easy to convert to parse_kv if you think it would be useful.

Also to me I can see this being in core or a collection.  I really don't care but I thought it was so basic that core would be appropriate.  I can move to `community.general` if you want.